moped-editor/src/views/projects/projectView/ProjectActivityLog.js

```diff
@@ -117,38 +116,70 @@ const ProjectActivityLog = () => {
    * @param {string} field - The field name (column name)
    * @return {boolean} - True if the field is contained in the ProjectActivityLogGenericDescriptions object
    */
-  const isFieldGeneric = field =>
+  const isFieldGeneric = (field) =>
     field in ProjectActivityLogGenericDescriptions;
 
   /**
-   * Makes sure the creation of the project always comes first
+   * if event is from new schema, finds the updated difference and includes in the object
+   * Makes sure the project creation event is the last in the returned Array
    * @param {Array} eventList - The data object as provided by apollo
-   * @return {Array}
+   * @returns {Array}
    */
-  const reorderCreationEvent = eventList => {
-    // Clone eventList array so it can be mutated
-    let outputList = [...eventList];
+  const getDiffs = (eventList) => {
+    let outputList = [];
+    let createdEvent = {};
+    eventList.forEach((event) => {
+      let outputEvent = { ...event };
+      // if the description includes "newSchema", we need to manually find the difference in the update
+      if (event.description[0]?.newSchema) {
+        // if event is an INSERT there is no previous record to compare to
+        if (event.operation_type === "INSERT") {
+          outputEvent.description = [];
+        } else {
+          // otherwise compare the old record to the new record to find the field that was updated
+          const newData = outputEvent.record_data.data.new;
+          const oldData = outputEvent.record_data.data.old;
+          let changedField = "";
+          Object.keys(newData).forEach((key) => {
+            if (newData[key] !== oldData[key]) {
+              changedField = key;
+            }
+          });
+          outputEvent.description = [
+            {
+              new: newData[changedField],
+              old: oldData[changedField],
+              field: changedField,
+            },
+          ];
+        }
+      }
+      outputList.push(outputEvent);
 
-    outputList.forEach(event => {
-      // If this is the creation of a project
+      // If this is the creation of a project, make a copy of it
+      // so we can make sure it shows up at the bottom of the list
       if (
-        event.record_type === "moped_project" &&
-        event.operation_type === "INSERT"
+        outputEvent.record_type === "moped_project" &&
+        outputEvent.operation_type === "INSERT"
       ) {
-        // Remove that object from the array and add it back on at the end
-        outputList.splice(outputList.indexOf(event), 1);
-        outputList.push(event);
+        createdEvent = outputEvent;
```

Comment:
turned out this is needed, after testing locally with prod data I found that many projects have the status creation or if its a signal project, the proj_component event is listed as happening before project creation.